tests/unit/amo/components/TestAddon.js

```diff
@@ -1374,4 +1374,44 @@ describe('mapStateToProps', () => {
 
     expect(addon).toEqual(null);
   });
+
+  // Test case for twitter card
+  it('adds twitter card information', () => {
+    const helmet = Helmet.peek();
+    const allMetaTags = helmet.metaTags;
+    let twitterMetaTitle;
+    for (const metaTag of allMetaTags) {
+      if (metaTag.name === 'twitter:title') {
+        twitterMetaTitle = metaTag;
+      }
+    }
+    const addon = createInternalAddon(fakeAddon);
+    expect(twitterMetaTitle.content).toBe(addon.name);
+  });
+
+  it('verify twitter card description', () => {
+    const helmet = Helmet.peek();
+    const allMetaTags = helmet.metaTags;
+    let twitterMet;
+    for (const metaTag of allMetaTags) {
+      if (metaTag.name === 'twitter:description') {
+        twitterMet = metaTag;
+      }
+    }
+    const addon = createInternalAddon(fakeAddon);
+    expect(twitterMet.content).toBe(addon.summary);
+  });
+
+  it('verify twitter card icon', () => {
+    const helmet = Helmet.peek();
+    const allMetaTags = helmet.metaTags;
+    let twitterMeta;
+    for (const metaTag of allMetaTags) {
+      if (metaTag.name === 'twitter:image') {
+        twitterMeta = metaTag;
+      }
+    }
+    const addon = createInternalAddon(fakeAddon);
+    expect(twitterMeta.content).toBe(addon.icon_url);
+  });
```

Comment:
It is interesting but we can probably simplify these test cases by not using `Helmet`. Can we get the `meta` tags by using `root.find('meta')`? That's what we are doing for the `title`: https://github.com/mozilla/addons-frontend/blob/e226ef4ae518561927fb514a7083f86d1501841e/tests/unit/amo/components/TestAddon.js#L1299-L1304

I like your test cases though, we indeed need several test cases. Note that the test descriptions (in `it('...'`) should be valid sentences, and you don't need to write "it" because it is already done by the `it` function). An example is:

```js
it('adds a meta "twitter:name" tag', () => {
  // ...
});
```

The test case description becomes:

> it adds a meta "twitter:name" tag